lib/Controller/LoginController.php

```diff
@@ -305,6 +303,37 @@ public function login(int $providerId, string $redirectUrl = null) {
 		return new RedirectResponse($authorizationUrl);
 	}
 
+	/**
+	 * @param string $authorizationEndpoint
+	 * @param array $extraGetParameters
+	 * @return string
+	 */
+	private function buildAuthorizationUrl(string $authorizationEndpoint, array $extraGetParameters = []): string {
+		$parsedUrl = parse_url($authorizationEndpoint);
+
+		$urlWithoutParams =
+			(isset($parsedUrl['scheme']) ? $parsedUrl['scheme'] . '://' : '')
+			. ($parsedUrl['host'] ?? '')
+			. (isset($parsedUrl['port']) ? ':' . $parsedUrl['port'] : '')
+			. ($parsedUrl['path'] ?? '');
+
+		$queryParams = $extraGetParameters;
+		if (isset($parsedUrl['query'])) {
+			$params = explode('&', $parsedUrl['query']);
+			foreach ($params as $p) {
```

Comment:
Could also use https://www.php.net/manual/en/function.parse-str.php to clean up the code a bit :)

Reply:
:+1: Why making things simple when we can make them complex? :sweat_smile: 

The `buildAuthorizationUrl` is now in the `DiscoveryService`. There are some basic tests covering this method.